wire-ios-sync-engine/Source/Use cases/GetSelfUserVerificationStatusUseCase.swift

```diff
@@ -0,0 +1,72 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireCoreCrypto
+
+// sourcery: AutoMockable
+public protocol GetSelfUserVerificationStatusUseCaseProtocol {
+    func invoke() async throws -> (isMLSCertified: Bool, isProteusVerified: Bool)
+}
+
+public struct GetSelfUserVerificationStatusUseCase: GetSelfUserVerificationStatusUseCaseProtocol {
+
+    private let context: NSManagedObjectContext
+    private let coreCryptoProvider: CoreCryptoProviderProtocol
+
+    public init(
+        context: NSManagedObjectContext,
+        coreCryptoProvider: CoreCryptoProviderProtocol
+    ) {
+        self.context = context
+        self.coreCryptoProvider = coreCryptoProvider
+    }
+
+    public func invoke() async throws -> (isMLSCertified: Bool, isProteusVerified: Bool) {
+        var isMLSCertified = false
+
+        let (selfUserID, isProteusVerified) = await context.perform { () -> (MLSClientID?, Bool) in
+            guard let selfClient = ZMUser.selfUser(in: context).selfClient() else { return (.none, false) }
+            return (MLSClientID(userClient: selfClient), selfClient.verified)
```

Comment:
`selfClient.verified `is just for a selfClient, but we should provide a status for a selfUser in this useCase. I think we should check the status of the other clients.

Reply:
Fixed, please check again.